modules/idaithalam/src/main/java/io/virtualan/idaithalam/contract/IdaithalamExecutor.java

```diff
@@ -280,17 +285,30 @@ private static void generateFeatureFile(ExecutionClassloader classloader, ApiExe
         for(int i=0; i< items.size(); i++){
             MustacheFactory mf = new DefaultMustacheFactory();
             Mustache mustache = mf.compile("virtualan-contract.mustache");
-            //FileOutputStream outputStream = new FileOutputStream(path+"/feature/virtualan-contract."+i+".feature");
-            FileOutputStream outputStream = new FileOutputStream(path+"/feature/" + removeFileName(items.get(i).getJsonFileName())+".feature");
+//            items.get(i).getWorkflowItems().get(0).setHasHeaderParams(true);
+            FileOutputStream outputStream = new FileOutputStream(path+"/feature/virtualan-contract."+i+".feature"); //TODO oglas uncommented this, please check. Line below does not work.
+//            FileOutputStream outputStream = new FileOutputStream(path+"/feature/" + removeFileName(items.get(i).getJsonFileName())+".feature");
             StringWriter writer = new StringWriter();
             mustache.execute(writer, new FeatureFileMapping(getTitle(featureTitle, i, feature), items.get(i).getWorkflowItems())).flush();
             PrettyFormatter formatter = new PrettyFormatter();
-            String formattedFeature = formatter.format(writer.toString());
-            writer.close();
-            outputStream.write(formattedFeature.getBytes(Charset.forName("UTF-8")));
-            outputStream.close();
+            String formattedFeature = null;
```

Comment:
This code needs to be reviewed and these code is been working for a while :(

Reply:
I made a try catch to give it more stability.
I had one request that could not be formatted, and then nothing was written to the feature file (it was the data from /virtualservices).
That shows that the formatter can break without leaving a hint was is wrong.
Therefore I ensured that the method writes the feature file even if the formatter fails.